managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -3643,7 +3645,10 @@ public void testDoNotGetOffloadPoliciesMultipleTimesWhenTrimLedgers() throws Exc
         config.setLedgerOffloader(ledgerOffloader);
 
         ledger.internalTrimConsumedLedgers(Futures.NULL_PROMISE);
-        verify(ledgerOffloader, times(1)).getOffloadPolicies();
+        final LedgerOffloader finalLedgerOffloader = ledgerOffloader;
+        Awaitility.await().untilAsserted(() -> {
+            verify(finalLedgerOffloader, times(1)).getOffloadPolicies();
+        });
```

Comment:
```suggestion

```

Reply:
@congbobo184 Not stable, because it may call zero time.

Reply:
we only verify the method `internalTrimConsumedLedgers` can invoke getOffloadPolicies() onece, it is a sync method, so the times can't 0.
https://github.com/apache/pulsar/blob/c55be388ad769974a1fd16d43517f4cfda19aec0/managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java#L3647

https://github.com/apache/pulsar/blob/c55be388ad769974a1fd16d43517f4cfda19aec0/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L2519